core/src/main/scala/org/apache/spark/sql/rapids/tool/qualification/QualificationAppInfo.scala

```diff
@@ -504,18 +504,12 @@ class QualificationAppInfo(
   }
 
   private def getMlFuntions: Option[Seq[MLFunctions]] = {
-    stageIdToInfo.map { case (_, stageInfo) =>
-      val mlOps = checkMLOps(stageInfo.info.details)
-      if (mlOps.nonEmpty) {
-        stageInfo.mlOps = Some(mlOps)
-      }
+    val mlFunctions = stageIdToInfo.map { case ((appId, _), stageInfo) =>
```

Comment:
do a flatMap here will get rid of any Options == None so you get Iterable[MLFunctions] back, then below you don't need to do the map to mlFun.get

Reply:
Thank you! Updated it.